easybuild/toolchains/compiler/fujitsu.py

```diff
@@ -0,0 +1,88 @@
+##
+# Copyright 2014-2021 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Support for the Fujitsu compiler drivers (aka fcc, frt).
+
+The basic concept the same as for the Cray Programming Environment.
+
+:author: Miguel Dias Costa (National University of Singapore)
+"""
+import easybuild.tools.systemtools as systemtools
+from easybuild.tools.toolchain.compiler import Compiler, DEFAULT_OPT_LEVEL
+
+TC_CONSTANT_FUJITSU = 'Fujitsu'
+
+
+class FujitsuCompiler(Compiler):
+    """Generic support for using Fujitsu compiler drivers."""
+    TOOLCHAIN_FAMILY = TC_CONSTANT_FUJITSU
+
+    # compiler module name is lang (with version e.g. tcsds-1.2.31)
+    COMPILER_MODULE_NAME = ['lang']
+    COMPILER_FAMILY = TC_CONSTANT_FUJITSU
+
+    COMPILER_CC = 'fcc -Nclang'
+    COMPILER_CXX = 'FCC -Nclang'
+    COMPILER_F77 = 'frt'
+    COMPILER_F90 = 'frt'
+    COMPILER_FC = 'frt'
+
+    COMPILER_UNIQUE_OPTION_MAP = {
+        DEFAULT_OPT_LEVEL: 'O2 -Kfast',
+        'lowopt': 'O1',
+        'noopt': 'O0',
+        'opt': 'O3 -Kfast',
```

Comment:
`-Kfast` is actually `-O3` plus a bunch of other stuff, so planning to play it safe and change to:

```python
    DEFAULT_OPT_LEVEL: 'O2',
    'opt': 'Kfast',
```
In addition to the possibility of using `'opt': True` in easyconfigs, `-Kfast` will be explicitly enabled in some easyblocks, e.g. fftw, berkeleygw (, fortranpythonpackage?)